src/Hazelcast.Net.Tests/Serialization/Compact/GenericRecordTests.cs

```diff
@@ -0,0 +1,406 @@
+
+        // cat set the IGenericRecord value
+        await mapOfGenericRecord.SetAsync(0, rec0);
+
+        // can get the value as an IGenericRecord
+        var rec0A = await mapOfGenericRecord.GetAsync(0);
+        Assert.That(rec0A, Is.Not.Null);
+        Assert.That(rec0A, Is.InstanceOf<CompactGenericRecordBase>());
+        Assert.That(rec0A, Is.InstanceOf<CompactDictionaryGenericRecord>());
+        Assert.That(rec0A.GetString("string"), Is.EqualTo("hello"));
+
+        await mapOfGenericRecord.DisposeAsync();
+        await using var mapOfObject = await Client.GetMapAsync<int, object>("generic-record-map");
+
+        // can get the value as an object -  because we have no registration for
+        // the compact type "rec0", we fall back to generic record
+        var rec0B = await mapOfGenericRecord.GetAsync(0);
+        Assert.That(rec0B, Is.Not.Null);
+        Assert.That(rec0B, Is.InstanceOf<CompactGenericRecordBase>());
+        Assert.That(rec0B, Is.InstanceOf<CompactDictionaryGenericRecord>());
+        Assert.That(rec0B.GetString("string"), Is.EqualTo("hello"));
+
+        await mapOfObject.DisposeAsync();
+    }
+
+    [Test]
+    public void CoverGenericRecord()
+    {
+        var schema = SchemaBuilder.For("type-name")
+            .WithField("field-name-1", FieldKind.Boolean)
+            .WithField("field-name-2", FieldKind.Int8)
+            .WithField("field-name-3", FieldKind.String)
+            .Build();
+
+        var fieldValues = new Dictionary<string, object?>
+        {
+            {"field-name-1", true},
+            {"field-name-2", 123},
+            {"field-name-3", "hello"}
+        };
+
+        var rec = new CompactDictionaryGenericRecord(schema, fieldValues);
+
+        Assert.That(rec.GetFieldKind("field-name-1"), Is.EqualTo(FieldKind.Boolean));
+        Assert.That(rec.GetFieldKind("field-name-2"), Is.EqualTo(FieldKind.Int8));
+        Assert.That(rec.GetFieldKind("field-name-3"), Is.EqualTo(FieldKind.String));
+
+        var fieldNames = rec.FieldNames;
+        Assert.That(fieldNames.Count, Is.EqualTo(3));
+        Assert.That(fieldNames, Does.Contain("field-name-1"));
+        Assert.That(fieldNames, Does.Contain("field-name-2"));
+        Assert.That(fieldNames, Does.Contain("field-name-3"));
+
+        Assert.Throws<ArgumentNullException>(() => _ = new CompactDictionaryGenericRecord(null!, fieldValues));
+        Assert.Throws<ArgumentNullException>(() => _ = new CompactDictionaryGenericRecord(schema, null!));
+    }
+
+    [Test]
+    public void CoverGenericRecordBuilder()
+    {
+        var schema = SchemaBuilder.For("type-name")
+            .WithField("field-name-1", FieldKind.Boolean)
+            .WithField("field-name-2", FieldKind.Int8)
+            .WithField("field-name-3", FieldKind.String)
+            .Build();
+
+        Assert.Throws<ArgumentException>(() => _ = new CompactGenericRecordBuilder((string) null!));
```

Comment:
It would be nice to have some comment here `ArgumentException` is better for string empty case for non-.net people.

Reply:
you mean, comment in the test to explain why we don't get an `ArgumentNullException`?

Reply:
Yeap.